src/Elastic.Transport/Components/Pipeline/DefaultRequestPipeline.cs

```diff
@@ -167,63 +170,110 @@ public override void BadResponse<TResponse>(ref TResponse response, ApiCallDetai
 			response = _responseBuilder.ToResponse<TResponse>(data, exception, callDetails?.HttpStatusCode, null, s, m, callDetails?.ResponseBodyInBytes?.Length ?? -1, null, null);
 		}
 
-		if (response.ApiCallDetails is ApiCallDetails apiCallDetails)
-			apiCallDetails.AuditTrail = AuditTrail;
+		response.ApiCallDetails.AuditTrail = AuditTrail;
 	}
 
 	public override TResponse CallProductEndpoint<TResponse>(RequestData requestData)
 	{
-		using (var audit = Audit(HealthyResponse, requestData.Node))
-		using (var d = RequestPipelineStatics.DiagnosticSource.Diagnose<RequestData, ApiCallDetails>(
-			DiagnosticSources.RequestPipeline.CallProductEndpoint, requestData))
+		using var audit = Audit(HealthyResponse, requestData.Node);
+
+		if (audit is not null)
+			audit.PathAndQuery = requestData.PathAndQuery;
+
+		var activity = Activity.Current;
+		var isElasticClient = activity is not null && activity.GetCustomProperty("elastic.transport.client") is not null;
+
+		if (!isElasticClient)
```

Comment:
Just catching up here but I am not fully grasping why we have this special case here?

Reply:
We start a [more specific activity](https://github.com/elastic/elasticsearch-net/blob/main/src/Elastic.Clients.Elasticsearch/Client/ElasticsearchClient.cs#L137) in the client which knows it's a `db.system` span and where we have access to the URL template to avoid high cardinality span names. We decided to avoid an unnecessary transport span in such cases to keep the span depth lower. Instead, if we know a client has set its own span, we add tags as necessary. Only if transport is being used directly, without a client, do we create the span here. Does that make sense?